internal/cli/quickstarts.go

```diff
@@ -111,218 +96,112 @@ func downloadQuickstartCmd(cli *cli) *cobra.Command {
-				return err
-			}
-
-			qsSamplePath := path.Join(target, quickstart.Samples[0])
-			readme, err := loadQuickstartSampleReadme(qsSamplePath) // Some QS have non-markdown READMEs (eg auth0-python uses rst)
-
-			if err == nil {
-				cli.renderer.Markdown(readme)
-			} else {
-				cli.renderer.Infof("%s You might wanna check out the Quickstart sample README", ansi.Faint("Hint:"))
-			}
-
-			relativeQSSamplePath, err := relativeQuickstartSamplePath(qsSamplePath)
-			if err != nil {
-				return err
-			}
-
-			cli.renderer.Infof("%s Start with 'cd %s'", ansi.Faint("Hint:"), relativeQSSamplePath)
-
-			return nil
-		},
+		RunE: downloadQuickstart(cli, &inputs),
 	}
 
 	cmd.SetUsageTemplate(resourceUsageTemplate())
 
 	cmd.Flags().BoolVar(&cli.force, "force", false, "Skip confirmation.")
-
 	qsStack.RegisterString(cmd, &inputs.Stack, "")
 
 	return cmd
 }
 
-func downloadQuickStart(ctx context.Context, client *management.Client, target string, q auth0.Quickstart) error {
-	request, err := http.NewRequestWithContext(ctx, http.MethodGet, quickstartEndpoint, nil)
-	if err != nil {
-		return unexpectedError(err)
-	}
+func listQuickstarts(cli *cli) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		quickstarts, err := auth0.GetQuickstarts()
+		if err != nil {
+			return err
+		}
 
-	params := request.URL.Query()
+		cli.renderer.QuickstartList(quickstarts)
 
-	// FIXME(copland): Default to first item from list of samples.
-	// Eventually we should add a forced survey for
-	// user to select one if there are multiple.
-	params.Add("branch", q.Branch)
-	params.Add("repo", q.Repo)
-	params.Add("path", q.Samples[0])
+		return nil
+	}
+}
 
-	// These appear to be largely constant and refers
-	// to the GitHub username they're under.
-	params.Add("org", quickstartOrg)
-	params.Add("client_id", client.GetClientID())
+func downloadQuickstart(cli *cli, inputs *qsInputs) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := inputs.fromArgs(cmd, args, cli); err != nil {
+			return fmt.Errorf("failed to parse command inputs: %w", err)
+		}
 
-	// Callback URL, if not set, it will just take the default one.
-	callbackURL := quickstartDefaultCallbackURL
-	if list := client.GetCallbacks(); len(list) > 0 {
-		callbackURL = list[0]
-	}
-	params.Add("callback_url", callbackURL)
+		quickstartPath, pathExists, err := computeQuickstartPathFromClientName(inputs.Client.GetName())
+		if err != nil {
+			return fmt.Errorf("failed to compute the path where to download the quickstart sample: %w", err)
+		}
 
-	request.URL.RawQuery = params.Encode()
-	request.Header.Set("Content-Type", quickstartContentType)
+		if pathExists && !cli.force {
+			message := fmt.Sprintf(
+				"%s %s already exists.\n Directory contents will be replaced. Are you sure you want to proceed? ",
+				ansi.Yellow("WARNING:"),
```

Comment:
Do we have a renderer function for warnings?

Reply:
We do have `cli.renderer.Warnf()`, however that message needs to go inside the `prompt.Confirm()` func. There doesn't seem to be any function that simply returns a warning string to be used elsewhere. Perhaps I'm missing something? 🤔

Reply:
That's right, my bad.